tests/components/plex/test_config_flow.py

```diff
@@ -0,0 +1,265 @@
+"""Tests for Plex config flow."""
+from unittest.mock import MagicMock, Mock, patch, PropertyMock
+import plexapi.exceptions
+import requests.exceptions
+
+from homeassistant.components.plex import config_flow
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN, CONF_URL
+
+from tests.common import mock_coro
+
+
+class MockAvailableServer:
+    """Mock avilable server objects."""
+
+    def __init__(self, name, client_id):
+        """Initialize the object."""
+        self.name = name
+        self.clientIdentifier = client_id
+        self.provides = ["server"]
+
+
+class MockConnection:
+    """Mock a single account resource connection object."""
+
+    def __init__(self):
+        """Initialize the object."""
+        self.httpuri = "http://1.2.3.4:32400"
+        self.uri = "http://4.3.2.1:32400"
+        self.local = True
+
+
+class MockConnections:
+    """Mock a list of resource connections."""
+
+    def __init__(self):
+        """Initialize the object."""
+        self.connections = [MockConnection()]
+
+
+async def test_bad_credentials(hass, aioclient_mock):
+    """Test when provided credentials are rejected."""
+
+    result = await hass.config_entries.flow.async_init(
```

Comment:
We usually instantiate the flow of our integration directly, set `hass` on it and then await the step we want to test. Calling the `config_entries` method also works and will load the integration and install requirements and dependencies, so it's a bit more that will happen. We might not need that for the tests, but it's not bad per se, I think?

Reply:
I'm open to whatever is considered best practice. Let me know which method you'd prefer.

Reply:
I think it's ok like this. Until someone presents a reason to one way or the other, let's keep it.